packages/replay/src/main.ts

```diff
@@ -466,18 +462,14 @@ async function uploadAllRecordings(opts: Options & FilterOptions = {}) {
   const server = getServer(opts);
   const dir = getDirectory(opts);
   const recordings = filterRecordings(readRecordings(dir), opts.filter);
-  let uploadedAll = true;
-
-  for (const recording of recordings) {
-    if (!uploadSkipReason(recording)) {
-      if (
-        !(await doUploadRecording(dir, server, recording, opts.verbose, opts.apiKey, opts.agent))
-      ) {
-        uploadedAll = false;
-      }
-    }
-  }
-  return uploadedAll;
+
+  const recordingIds = await Promise.allSettled(
+    recordings
+      .filter(r => !uploadSkipReason(r))
+      .map(r => doUploadRecording(dir, server, r, opts.verbose, opts.apiKey, opts.agent))
```

Comment:
We should probably have some maximum on the amount of parallelism here

Reply:
Grouping into 10-20 sounds great. 

The other concern here that I see is that `connectionUploadRecording` is currently written in such a way that the entire recording file could potentially be loaded into memory nearly all at once, so running this in parallel with many large recordings could put this at risk of using up all of Node's memory. We may want to consider updating `connectionUploadRecording` to read data from disk at the rate that it is able to flush the commands from the network buffers

Reply:
It looks like your change went from having a single Buffer, to having many smaller buffers, but there's nothing that controls the _rate_ of the data being read from disk, so the most recording will likely be read into memory, since the machine will be able to read from disk a lot faster than it can send the data over the network.

What I'm saying is that ideally `gClient?.sendCommand("Internal.addRecordingData", { recordingId, offset, length }, b)` would ideally also take some extra callback parameter that would be called when the data has been flushed from the network buffers, so you'd for instance load 1MB from disk, send that, wait for it to flush from the network buffer, then read another 1MB, that way no matter how large the recording is, you only have 1MB of a given recording in memory at once.

Reply:
In the case of this codebase, https://github.com/replayio/replay-cli/blob/391a37c41488d5f5231799152f7bdd187ea10e0d/packages/replay/src/client.ts#L62 the `send` function takes a callback as its second argument https://github.com/websockets/ws/blob/HEAD/doc/ws.md#websocketsenddata-options-callback